c/tskit/genotypes.c

```diff
@@ -623,7 +617,75 @@ tsk_vargen_update_site(tsk_vargen_t *self)
          * not-missing */
         num_missing -= (tsk_size_t) no_longer_missing;
     }
-    var->has_missing_data = num_missing > 0;
+    self->has_missing_data = num_missing > 0;
+out:
+    return ret;
+}
+
+int
+tsk_tree_get_variant(tsk_tree_t *self, const tsk_site_t *site, tsk_variant_t *variant,
```

Comment:
Do we need this method for the first pass? We just want to refactor the vargen for now, keeping the same semantics and performance as before, right?